plugins/window_size/linux/src/window_size_plugin.cc

```diff
@@ -0,0 +1,221 @@
+  WindowSizePlugin(
+      std::unique_ptr<flutter::MethodChannel<EncodableValue>> channel,
+      flutter::FlutterWindow *window);
+
+  // Called when a method is called on |channel_|;
+  void HandleMethodCall(
+      const flutter::MethodCall<EncodableValue> &method_call,
+      std::unique_ptr<flutter::MethodResult<EncodableValue>> result);
+
+  // The MethodChannel used for communication with the Flutter engine.
+  std::unique_ptr<flutter::MethodChannel<EncodableValue>> channel_;
+
+  // The Flutter window.
+  flutter::FlutterWindow *window_;
+};
+
+// static
+void WindowSizePlugin::RegisterWithRegistrar(
+    flutter::PluginRegistrarGlfw *registrar) {
+  auto channel = std::make_unique<flutter::MethodChannel<EncodableValue>>(
+      registrar->messenger(), kChannelName,
+      &flutter::StandardMethodCodec::GetInstance());
+  auto *channel_pointer = channel.get();
+
+  // Uses new instead of make_unique due to private constructor.
+  std::unique_ptr<WindowSizePlugin> plugin(
+      new WindowSizePlugin(std::move(channel), registrar->window()));
+
+  channel_pointer->SetMethodCallHandler(
+      [plugin_pointer = plugin.get()](const auto &call, auto result) {
+        plugin_pointer->HandleMethodCall(call, std::move(result));
+      });
+  registrar->EnableInputBlockingForChannel(kChannelName);
+
+  registrar->AddPlugin(std::move(plugin));
+}
+
+WindowSizePlugin::WindowSizePlugin(
+    std::unique_ptr<flutter::MethodChannel<EncodableValue>> channel,
+    flutter::FlutterWindow *window)
+    : channel_(std::move(channel)), window_(window) {}
+
+WindowSizePlugin::~WindowSizePlugin() {}
+
+void WindowSizePlugin::HandleMethodCall(
+    const flutter::MethodCall<EncodableValue> &method_call,
+    std::unique_ptr<flutter::MethodResult<EncodableValue>> result) {
+  if (method_call.method_name().compare(kGetScreenListMethod) == 0) {
+    EncodableValue screens(EncodableValue::Type::kList);
+    GdkDisplay *display = gdk_display_get_default();
+    if (!display) {
+      result->Error("Unable to get display");
+      return;
+    }
+    int monitor_count = gdk_display_get_n_monitors(display);
+    for (int i = 0; i < monitor_count; ++i) {
+      GdkMonitor *monitor = gdk_display_get_monitor(display, i);
+      screens.ListValue().push_back(
+          GetPlatformChannelRepresentationForMonitor(monitor));
+    }
+    result->Success(&screens);
+  } else if (method_call.method_name().compare(kGetWindowInfoMethod) == 0) {
+    EncodableValue window_info =
+        GetPlatformChannelRepresentationForWindow(window_);
+    result->Success(&window_info);
+  } else if (method_call.method_name().compare(kSetWindowFrameMethod) == 0) {
+    if (!method_call.arguments() || !method_call.arguments()->IsList() ||
+        method_call.arguments()->ListValue().size() != 4) {
+      result->Error("Bad arguments", "Expected 4-element list");
+      return;
+    }
+    const auto &frame_list = method_call.arguments()->ListValue();
```

Comment:
Should we sanity check the arguments here? E.g. negative width/height disallowed

Reply:
Ah. In that case, I'd suggest documenting the assumption made here that the sanity checking is already done (otherwise it might be tempting for some future reader to "fix" things by adding code here).

Reply:
Done.